File: backend-v5/lms/poll/PollQuestion.php
<?php

// Modul: lms/poll
// Klasa: PollQuestion
// Opis: pitanje na nekoj anketi

// TODO ovo treba preurediti tako da postoji many-to-many relacija između ankete i pitanja, tako da se isto pitanje može ponavljati na više anketa... onda treba skontati kako to u GUIju riješiti

require_once(Config::$backend_path."core/DB.php");

class PollQuestion {
	public $id;
	public $pollId, $typeId /* TODO preći na enum */, $text;

	public static function fromId($id) {
		$q10 = DB::query("select anketa, tip_pitanja, tekst from anketa_pitanje where id=$id");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("unknown poll question");
		}
		
		$pq = new PollQuestion;
		$pq->id = $id;
		$pq->pollId = mysql_result($q10,0,0);
		$pq->typeId = mysql_result($q10,0,1);
		$pq->text = mysql_result($q10,0,2);

		return $pq;
	}

	// Get all questions in a poll
	public static function getAllForPoll($pollId) {
		$q10 = DB::query("select id, tip_pitanja, tekst from anketa_pitanje where anketa=$pollId order by id"); 
		// It's assumed that questions are ordered by id in the order that they are supposed to appear in a poll - FIXME there should be an ordering value
		$questions = array();
		while ($r10 = mysql_fetch_row($q10)) {
			$pq = new PollQuestion;
			$pq->id = $r10[0];
			$pq->pollId = $pollId;
			$pq->typeId = $r10[1];
			$pq->text = $r10[2];
			
			array_push($questions, $pq);
		}
		return $questions;
	}
	
	public function setAnswerRank($pollResultId, $response) {
		if ($this->typeId != 1) return;
		if ($response > 0) // 0 means "no response" and we just won't insert it into database
			$q10 = DB::query("insert into anketa_odgovor_rank set rezultat=$pollResultId, pitanje=".$this->id.", izbor_id=$response");
	}
	
	public function setAnswerEssay($pollResultId, $response) {
		if ($this->typeId != 2) return;
		if (preg_match("/\w/", $response))  // Skip blank comments
			$q10 = DB::query("insert into anketa_odgovor_text set rezultat=$pollResultId, pitanje=".$this->id.", odgovor='$response'");
	}
	
	public function setAnswerChoice($pollResultId, $response, $writein) {
		if ($this->typeId != 3 && $this->typeId != 4) return;
		
		// Does this answer allow writein response
		$q10 = DB::query("select dopisani_odgovor from anketa_izbori_pitanja where pitanje=".$this->id." and id=$response");
		if (mysql_result($q10,0,0)==1) {
			$q20 = myquery("insert into anketa_odgovor_dopisani set rezultat=$pollResultId, pitanje=".$this->id.", odgovor='$writein'");
		}
		$q30 = DB::query("insert into anketa_odgovor_izbori set rezultat=$pollResultId, pitanje=".$this->id.", izbor_id=$response");
	}
}

?>

File: rorbackend/app/models/lms/poll/poll_question.rb
class Lms::Poll::PollQuestion < ActiveRecord::Base
  # Uncomment following lines if working with legacy database
  # set_table_name 'anketa_pitanje'
  # set_primary_key :id
  # alias_attribute :id, :id
  # alias_attribute :poll_id, :anketa
  # alias_attribute :poll_question_type_id, :tip_pitanja
  # alias_attribute :text, :tekst

  # Uncomment following lines if working with legacy database
  # TABLE_NAME = 'anketa_pitanje'
  # ID = TABLE_NAME + '.' + 'id'
  # POLL_ID = TABLE_NAME + '.' + 'anketa'
  # POLL_QUESTION_TYPE_ID = TABLE_NAME + '.' + 'tip_pitanja
  # TEXT = TABLE_NAME + '.' + 'tekst'
 
  # Comment following lines if working with legacy database
  TABLE_NAME = 'lms_poll_poll_questions'
  ID = TABLE_NAME + '.' + 'id'
  POLL_ID = TABLE_NAME + '.' + 'poll_id'
  POLL_QUESTION_TYPE_ID = TABLE_NAME + '.' + 'poll_question_type_id'
  TEXT = TABLE_NAME + '.' + 'text'

  ALL_COLUMNS = [ID, POLL_ID, POLL_QUESTION_TYPE_ID, TEXT]
  
  belongs_to :poll
  has_one :poll_question_type
  
  
  def self.get_all_for_poll(poll_id)
    poll_questions = (Lms::Poll::PollQuestion).where(:poll_id => poll_id)
    return poll_questions
  end
  
  
  def self.set_answer_rank(type_id, response, id, poll_result_id, poll_question_choice_id)
    raise ArgumentError if (type_id != 1)
    
    if (response > 0)
      answer_rank = (Lms::Poll::PollAnswerRank).new(:poll_result_id => poll_result_id, :poll_question_id => id, :poll_question_choice_id => poll_question_choice_id)
      return answer_rank.save
    end
  end
  
  def self.set_answer_essay(type_id, id, poll_result_id, answer)
    raise ArgumentError if (type_id != 1)
    
    answer_essay = (Lms::Poll::PollAnswerEssay).new(:poll_result_id => poll_result_id, :poll_question_id => id, :answer => answer)
    
    return answer_essay.save
  end
  
end
